common/config/rpc.go

```diff
@@ -90,20 +112,44 @@ func (d *RPCFactory) CreateDispatcherForOutbound(
 	callerName string,
 	serviceName string,
 	hostName string,
-) *yarpc.Dispatcher {
+) (*yarpc.Dispatcher, error) {
+	return d.createOutboundDispatcher(callerName, serviceName, hostName, d.ch.NewSingleOutbound(hostName))
+}
+
+// CreateGRPCDispatcherForOutbound creates a dispatcher for GRPC outbound connection
+func (d *RPCFactory) CreateGRPCDispatcherForOutbound(
+	callerName string,
+	serviceName string,
+	hostName string,
+) (*yarpc.Dispatcher, error) {
+	grpcAddress, err := d.grpcPorts.GetGRPCAddress(serviceName, hostName)
```

Comment:
Are we assuming that all instances use the same grpc port on different hosts?

Reply:
Yes, at least for open source version here.